joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -622,6 +616,22 @@ CallbackReturn JointTrajectoryController::on_configure(const rclcpp_lifecycle::S
       "interfaces are not present.");
     return CallbackReturn::FAILURE;
   }
+  else if (has_velocity_command_interface_ && command_interface_types_.size() == 1)
```

Comment:
This checks here are a bit hard to understand. And it could happen that they pass if “acceleration state interface” is present. (This is also not explicitly tested.) It seems to me that there are too many checks with “else if”.

What do you think? Would it be better to separate checks? Then the code would be cleaner and functionality should be preserved, e.g.:
```suggestion
  if (has_velocity_command_interface_ && command_interface_types_.size() == 1)
```

Reply:
Just to clarify, should we allow or not allow the acceleration state interface when using effort-only or velocity-only command interfaces?

Reply:
We can allow it, but I think that in this case if acceleration interface is there, other combination will not be checked. Maybe I am misunderstanding something (?).